### PR TITLE
feat(app-notifications): Customize app notifications.

### PR DESCRIPTION
**Pull request by tudor-phd**

### CODE REVIEW

File: modules/UI/util/MessageHandler.js [hunk 1]
```diff
@@ -117,7 +117,7 @@ const messageHandler = {
             cls,
             messageKey,
             messageArguments,
-            timeout = NOTIFICATION_TIMEOUT) {
+            timeout = NOTIFICATION_TIMEOUT_TYPE.SHORT) {
```
Comment (saghul) [inline modules/UI/util/MessageHandler.js:120]: Can you please go ahead and remove this `participantNotification` altogether? It's only used by UI.js which is then called by conference.js, so we should be able to remove all that legacy code and dispatch the notifications directly from conference.js

File: react/features/base/participants/actions.js [hunk 2]
```diff
@@ -510,7 +510,7 @@ export function participantKicked(kicker, kicked) {
                     getParticipantDisplayName(getState, kicker.getId())
             },
             titleKey: 'notify.kickParticipant'
-        }, NOTIFICATION_TIMEOUT * 2)); // leave more time for this
+        }, NOTIFICATION_TIMEOUT_TYPE.MEDIUM)); // leave more time for this
```
Comment (saghul) [inline react/features/base/participants/actions.js:513]: Drop the comment please.

File: react/features/lobby/middleware.js [hunk 3]
```diff
@@ -9,7 +9,11 @@ import { getFirstLoadableAvatarUrl, getParticipantDisplayName } from '../base/pa
 import { MiddlewareRegistry, StateListenerRegistry } from '../base/redux';
 import { playSound, registerSound, unregisterSound } from '../base/sounds';
 import { isTestModeEnabled } from '../base/testing';
-import { NOTIFICATION_TYPE, showNotification } from '../notifications';
+import {
+    NOTIFICATION_TYPE,
+    showNotification,
+    NOTIFICATION_TIMEOUT_TYPE
```
Comment (saghul) [inline react/features/lobby/middleware.js:15]: Group the capitalized constants at the top please.

File: react/features/lobby/middleware.js [hunk 4]
```diff
@@ -217,5 +230,10 @@ function _maybeSendLobbyNotification(origin, message, { dispatch, getState }) {
         break;
     }
 
-    dispatch(showNotification(notificationProps, isTestModeEnabled(getState()) ? undefined : 5000));
+    dispatch(
+        showNotification(
+            notificationProps,
+            isTestModeEnabled(getState()) ? undefined : NOTIFICATION_TIMEOUT_TYPE.MEDIUM
```
Comment (saghul) [inline react/features/lobby/middleware.js:236]: We should never pass undefined as a timeout.

File: react/features/notifications/actions.js [hunk 5]
```diff
@@ -14,11 +14,34 @@ import {
...
+declare var APP: Object;
+
+/**
+ * Function that returns notification timeout value based on notification timeout type.
+ *
+ * @param {string} type - Notification type.
+ * @returns {number}
+ */
+function getNotificationTimeout(type: ?string) {
+    const { notificationTimeouts } = APP.store.getState()['features/base/config'];
```
Comment (saghul) [inline react/features/notifications/actions.js:32]: This is not going to work on mobile, you can't access APP there. You should get a reference to the state as a  parameter and extract the timeouts from there.

File: react/features/notifications/components/web/NotificationsContainer.js [hunk 6]
```diff
@@ -290,14 +289,15 @@ class NotificationsContainer extends Component<Props> {
...
+    const { notificationTimeouts } = state['features/base/config'];
     const { isOpen: isChatOpen } = state['features/chat'];
     const _visible = areThereNotifications(state);
 
     return {
         _iAmSipGateway: Boolean(iAmSipGateway),
         _isChatOpen: isChatOpen,
         _notifications: _visible ? notifications : [],
-        autoDismissTimeout: interfaceConfig.ENFORCE_NOTIFICATION_AUTO_DISMISS_TIMEOUT
+        autoDismissTimeout: notificationTimeouts?.long ?? NOTIFICATION_TIMEOUT.LONG
```
Comment (saghul) [inline react/features/notifications/components/web/NotificationsContainer.js:300]: Get rid of this altogether, the notification will now always have a timeout set.

File: react/features/notifications/constants.js [hunk 7]
```diff
@@ -3,7 +3,22 @@
 /**
  * The standard time when auto-disappearing notifications should disappear.
  */
-export const NOTIFICATION_TIMEOUT = 2500;
+export const NOTIFICATION_TIMEOUT = {
+    SHORT: 2500,
+    MEDIUM: 5000,
+    LONG: 10000,
+    STICKY: 'sticky'
```
Comment (saghul) [inline react/features/notifications/constants.js:10]: Where are you checking for this in order to avoid creating the timer?

**Review by saghul — CHANGES_REQUESTED** [review]
This is looking good! Left some comments, PTAL!

File: react/features/notifications/constants.js (hunk 7 above)
Comment (tudor-phd) [reply react/features/notifications/constants.js]: Changed STICKY value to Infinity.

**Comment by saghul** [comment]
Jenkins please test this please.

**Review by saghul — APPROVED** [review]

**Pull request merged**